File: Go/01.Go-Basic-February-2022/10.advancedLoops-Exercise/01.oldBooks.go
package main

import "fmt"

func main() {
	var searchedBook string
	fmt.Scanln(&searchedBook)

	var inputBook string
	fmt.Scanln(&inputBook)

	var count int = 0
	var isFound bool = false
	for inputBook != "NoMoreBooks" {
		if inputBook == searchedBook {
			fmt.Printf("You checked %d books and found it.", count)
			isFound = true
			break
		}
		count++
		fmt.Scanln(&inputBook)
	}
	
	if !isFound {
		fmt.Println("The book you search is not here!")
		fmt.Printf("You checked %d books.", count)
	}
}

File: JavaScript/01.JavaScript-Basic-February-2021/10.whileLoop-Exersice/01.oldBooks.js
function oldBooks(input) {
    let favoriteBook = input[0];

    let index = 1;
    let bookIsFound = false;

    let nextBookName = input[index];
    while (nextBookName !== "No More Books") {
        if (nextBookName === favoriteBook) {
            bookIsFound = true;
            break;
        }
        index++;
        nextBookName = input[index];
    }

    if (bookIsFound === false) {
        console.log(`The book you search is not here!`);
        console.log(`You checked ${index - 1} books.`);
    } else {
        console.log(`You checked ${index - 1} books and found it.`);
    }
}

oldBooks(["Troy", "Stronger", "Life Style", "Troy"]);
oldBooks(["The Spot", "Hunger Games", "Harry Potter", "Torronto", "Spotify", "No More Books"]);
oldBooks(["Bourne", "True Story", "Forever", "More Space", "The Girl", "Spaceship", "Strongest", "Profit", "Tripple", "Stella", "The Matrix", "Bourne"]);